components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/CacheBackedExternalClaimDAO.java

```diff
@@ -72,20 +75,32 @@ public void addExternalClaim(ExternalClaim externalClaim, int tenantId) throws C
         String externalClaimDialectURI = externalClaim.getClaimDialectURI();
         ExternalClaimCacheKey cacheKey = new ExternalClaimCacheKey(externalClaimDialectURI);
         externalClaimCache.clearCacheEntry(cacheKey, tenantId);
+        associatedClaimCache.clearCacheEntry(externalClaim.getMappedLocalClaim(), tenantId);
     }
     public void updateExternalClaim(ExternalClaim externalClaim, int tenantId) throws ClaimMetadataException {
 
         externalClaimDAO.updateExternalClaim(externalClaim, tenantId);
         String externalClaimDialectURI = externalClaim.getClaimDialectURI();
         ExternalClaimCacheKey cacheKey = new ExternalClaimCacheKey(externalClaimDialectURI);
         externalClaimCache.clearCacheEntry(cacheKey, tenantId);
+        associatedClaimCache.clearCacheEntry(externalClaim.getMappedLocalClaim(), tenantId);
     }
     public void removeExternalClaim(String externalClaimDialectURI, String externalClaimURI, int tenantId) throws
             ClaimMetadataException {
 
+        List<ExternalClaim> externalClaimsList = getExternalClaims(externalClaimDialectURI, tenantId);
```

Comment:
If `externalClaimsList` is nullable, check not null before looping

Reply:
Fixed